### PR TITLE
vertexai: Support list URI input to `embed_image()` API

### PR DESCRIPTION
**Pull request by duwenxin99**
- Other embedding services are supporting list of URI input to the image embedding API `embed_image()`. Replacing the current `embed_image()` API with `embed_images()` that takes a list of image URIs as input allows it to be used with the [LangChain vector store and similarity search integrations](https://github.com/googleapis/langchain-google-alloydb-pg-python/pull/207) that are currently being built for Google Cloud databases.

### CODE REVIEW

File: libs/vertexai/langchain_google_vertexai/embeddings.py [hunk 1]
```diff
@@ -470,27 +470,31 @@ def embed_query(self, text: str) -> List[float]:
 
     def embed_image(
         self,
-        image_path: str,
+        uris: List[str],
```
Comment (lkuligin) [inline libs/vertexai/langchain_google_vertexai/embeddings.py:473]: it seems like a breaking change to me, can we add a deprecation warning instead and implement a new method maybe?

File: libs/vertexai/langchain_google_vertexai/embeddings.py [hunk 2]
```diff
@@ -470,27 +470,31 @@ def embed_query(self, text: str) -> List[float]:
...
         image_loader = ImageBytesLoader()
-        bytes_image = image_loader.load_bytes(image_path)
-        image = Image(bytes_image)
-        result: MultiModalEmbeddingResponse = self.instance[
-            "get_embeddings_with_retry"
-        ](image=image, contextual_text=contextual_text, dimension=dimensions)
-        return result.image_embedding
+        embeddings = []
+        for image_path in uris:
+            bytes_image = image_loader.load_bytes(image_path)
```
Comment (lkuligin) [inline libs/vertexai/langchain_google_vertexai/embeddings.py:493]: do embeddings models support pointers to GCS actually?

**Review by lkuligin — CHANGES_REQUESTED** [review]

File: libs/vertexai/langchain_google_vertexai/embeddings.py (hunk 2 above)
Comment (duwenxin99) [reply libs/vertexai/langchain_google_vertexai/embeddings.py:493]: Currently other embedding models mostly only support local URIs.

File: libs/vertexai/langchain_google_vertexai/embeddings.py (hunk 1 above)
Comment (duwenxin99) [reply libs/vertexai/langchain_google_vertexai/embeddings.py]: added deprecation warning and new method

File: libs/vertexai/langchain_google_vertexai/embeddings.py [hunk 3]
```diff
@@ -498,5 +504,33 @@ def embed_image(
         ](image=image, contextual_text=contextual_text, dimension=dimensions)
         return result.image_embedding
 
+    def embed_images(
+        self,
+        uris: List[str],
+        contextual_text: Optional[str] = None,
+        dimensions: Optional[int] = None,
+    ) -> List[float]:
```
Comment (lkuligin) [inline libs/vertexai/langchain_google_vertexai/embeddings.py:512]: should it be List[List[float]] maybe?

Comment (duwenxin99) [reply libs/vertexai/langchain_google_vertexai/embeddings.py]: Thanks for catching that. Updated.

**Review by lkuligin — APPROVED** [review]

File: libs/vertexai/langchain_google_vertexai/embeddings.py [hunk 4]
```diff
@@ -487,6 +488,12 @@ def embed_image(
         Returns:
             Embedding for the image.
         """
+        warnings.warn(
```
[the remark this answers is not in the document]
Comment (duwenxin99) [reply libs/vertexai/langchain_google_vertexai/embeddings.py:496]: Added. Is there a specific removal version you'd like?

**Review by lkuligin — APPROVED** [review]

File: libs/vertexai/langchain_google_vertexai/embeddings.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (lkuligin) [reply libs/vertexai/langchain_google_vertexai/embeddings.py:496]: all good, thx!

**Pull request merged**